### DISCUSSION TITLE
Password updates for users

### DISCUSSION THREAD

**Discussion by MarcS1975** [docker-mailserver/docker-mailserver#3745] [Ideas]
I am looking for a way to let users update their own passwords without sys admin involvement. Is there a way?
The topic has been raised a few times and closed again.
E.g. I am using roundcube as front end mail client and there is a password plugin that is described here: [roundcube password plugin](https://stackoverflow.com/questions/62655236/how-to-enable-password-plugin-on-roundcube) but this is of course not integrated with DMS.
Does anyone have a workable solution?

**Comment by polarathene** [comment]
> I am looking for a way to let users update their own passwords without sys admin involvement. Is there a way?
> The topic has been raised a few times and closed again.

You could use LDAP and any software that provides that self-service feature.

The file-based accounts is rather basic and subject to change in future. There was some talk of efforts to build an API around it, but those appear to have become stale.

---

> E.g. I am using roundcube as front end mail client and there is a password plugin that is described here: [roundcube password plugin](https://stackoverflow.com/questions/62655236/how-to-enable-password-plugin-on-roundcube) but this is of course not integrated with DMS.

I recall an issue or discussion about this. I think someone tried to integrate that with DMS, you'd need to dig that up to find out if they were successful or ping them for an update.

---

> Does anyone have a workable solution?

If you find the longer issues regarding this feature request, one does chime in with their own Github project that offered support for updating DMS accounts via an API IIRC. It might meet your needs?

**Comment by MarcS1975** [comment]
Thanks! 
LDAP is too cumbersome. I only have a hand full of users but its not practical to let them use ssh in order to update their own password.
I will keep looking...

**Reply by polarathene** [reply]
We will be merging a feature soon for OAUTH2 support.

It still uses LDAP or FILE account setup, but for users that only need to access DMS from a mail client that supports OAUTH2 login (Roundcube does), they can use that instead to authenticate.

The contributor of the feature used Authentik (GUI focused) with Roundcube to use this feature. It might work for you, the feature comes with a docs guide for using Authentik + Roundcube with the DMS feature, but doesn't hand-hold Authentik + Roundcube setup.

The feature will be merged soon, and part of DMS v14 that will likely be released this month.

**Reply by polarathene** [reply]
Just to clarify, that won't help with changing the account password. But if you have them sign in with OAUTH2, then whatever that third-party account provider is (_Google, Github, your own, etc_), then the user can change their password for that unrelated to DMS.

They just get redirected from Roundcube to the identity provider and perform a login to continue 👍 (_there needs to be a matching account created in DMS though, which is typically the email address_)

**Comment by audioscavenger** [comment]
if you are patient enough, the dms-gui that i am working on will let you do that, by creating extra users that can login and assign roles to act on their own accounts
